test/recipes/70-test_certtypeext.t

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/env perl
+# Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use OpenSSL::Test qw/:DEFAULT cmdstr srctop_file bldtop_dir/;
+use OpenSSL::Test::Utils;
+use TLSProxy::Proxy;
+
+my $test_name = "test_certtypeext";
+setup($test_name);
+
+plan skip_all => "TLSProxy isn't usable on $^O"
+    if $^O =~ /^(VMS)$/;
+
+plan skip_all => "$test_name needs the dynamic engine feature enabled"
+    if disabled("engine") || disabled("dynamic-engine");
+
+plan skip_all => "$test_name needs the sock feature enabled"
+    if disabled("sock");
+
+plan skip_all => "$test_name needs TLS enabled"
+    if alldisabled(available_protocols("tls"))
+       || (!disabled("tls1_3") && disabled("tls1_2"));
+
+my $proxy = TLSProxy::Proxy->new(
+    \&certtype_filter,
+    cmdstr(app(["openssl"]), display => 1),
+    srctop_file("apps", "server.pem"),
+    (!$ENV{HARNESS_ACTIVE} || $ENV{HARNESS_VERBOSE})
+);
+
+use constant {
+    SERVER_CERT_TYPE => 0,
+    CLIENT_CERT_TYPE => 1,
+    NO_CERT_TYPE => 2
+};
+my $testtype;
+plan tests => 4;
+
+# Test 1: Just do a verify without cert type
+$proxy->clear();
+$proxy->clientflags("-tls1_2 -cert ".srctop_file("apps", "server.pem"));
+$proxy->serverflags("-verify 4");
+$testtype = NO_CERT_TYPE;
+$proxy->start();
```

Comment:
Change to this:

```` perl
$proxy->start() or plan skip_all => "Unable to start up Proxy for tests";
plan tests => 4;
````

We have historically had some problems on some corner case platforms which we weren't able to support but don't detect until we attempt to start the proxy for the first time. We skip all the tests if the proxy fails to even start.

Reply:
OK, I saw that construct, but after changing the test a number of times, I ended up removing it.